packages/blockfrost/src/blockfrostProvider.ts

```diff
@@ -122,21 +151,32 @@ export const blockfrostProvider = (options: Options, logger = dummyLogger): Wall
     await blockfrost.txSubmit(signedTransaction);
   };
 
-  const utxoDelegationAndRewards: WalletProvider['utxoDelegationAndRewards'] = async (addresses, stakeKeyHash) => {
+  const utxoDelegationAndRewards: WalletProvider['utxoDelegationAndRewards'] = async (addresses, rewardAccount) => {
     const utxoResults = await Promise.all(
       addresses.map(async (address) =>
-        blockfrost.addressesUtxosAll(address).then((result) => BlockfrostToCore.addressUtxoContent(address, result))
+        fetchByAddressSequentially<Cardano.Utxo, BlockfrostUtxo>({
+          address,
+          request: (addr: Cardano.Address, pagination) => blockfrost.addressesUtxos(addr, pagination),
+          responseTranslator: (addr: Cardano.Address, response: Responses['address_utxo_content']) =>
+            BlockfrostToCore.addressUtxoContent(addr, response)
+        })
       )
     );
     const utxo = utxoResults.flat(1);
 
-    const accountResponse = await blockfrost.accounts(stakeKeyHash);
-    const delegationAndRewards = {
-      delegate: accountResponse.pool_id || undefined,
-      rewards: BigInt(accountResponse.withdrawable_amount)
-    };
-
-    return { delegationAndRewards, utxo };
+    try {
+      const accountResponse = await blockfrost.accounts(rewardAccount);
+      const delegationAndRewards = {
+        delegate: accountResponse.pool_id || undefined,
+        rewards: BigInt(accountResponse.withdrawable_amount)
+      };
+      return { delegationAndRewards, utxo };
+    } catch (error) {
+      if (error.status_code === 404) {
```

Comment:
Getting a ts error here since `error` is of type `unknown`

Reply:
Hmm that's odd, I didn't see the compiler complaining :thinking:

Reply:
`yarn test` works without errors for me too, but the ts error is shown by my vscode :thinking:

Reply:
Tried it. There must be a difference between our setups. Anyway I think this an actual issue to fix because we don't want to be using `unknown` type without explicitly casting it.